### PR TITLE
Feat/jwt to session based auth

### LINKED ISSUE #916: `0.7.0` Auth migration from JWT to session-based
Currently we are using JWT based authentication based system from which we would be now transition or migrate to session based authentication.

### PR DESCRIPTION
**Pull request by Mantu01**
## Description

closes #916 

This PR migrates authentication from **JWT-based authentication** to a **session-based authentication** model.

The existing stateless JWT flow has been replaced with server-managed sessions to improve security, simplify auth lifecycle management, and support upcoming features such as MFA and multi-session handling.

### Key Changes
- Replaced JWT-based auth with session-based authentication
- Added session creation, validation, and invalidation logic
- Updated auth middleware and controllers to rely on sessions
- Removed JWT-specific logic where no longer required
- Added comprehensive test coverage for the new auth flow

### Why this change?
- Better control over active user sessions
- Simplified logout and session invalidation
- Improved security by avoiding long-lived tokens
- Lays the foundation for MFA and multi-purpose verification sessions

### Testing
- Added unit and integration tests for:
  - Session creation and validation
  - Authenticated and unauthenticated flows
  - Edge cases previously handled via JWT
- All new and existing tests pass locally
- Coverage reports attached for reference

## What type of PR is this? (Check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [ ] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🛠️ CI/CD

## Checklist
- [x] I have performed a self-review of my code  
- [x] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes
Coverage
image
image

### CODE REVIEW

**Comment by MukeshAbhi** [comment]
@Mantu01 @Smayur0 Add PR description and also link this with an issue

File: packages/app/models/userSession.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    required: true,
+    unique: true,
+  },
+
+  isActive: {
+    type: Boolean,
+    default: true,
+  },
+
+  lastActivityAt: {
```
Comment (anujarora0502) [inline packages/app/models/userSession.js]: We are not using this anywhere, why did we add it? Are we planning to use it in the future?

**Review by anujarora0502 — COMMENTED** [review]
I have one query only, let me know on discord whenever you see it, I will approve it once it is resolved, thanks!

File: packages/app/models/userSession.js (hunk 1 above)
Comment (Smayur0) [reply packages/app/models/userSession.js]: ill remove this as of now

**Review by anujarora0502 — APPROVED** [review]
Thanks for addressing my concern, approving now!
